### PR TITLE
added intel edison support

### PR DESCRIPTION
**Pull request by alessandro-aglietti**
os_detect with this patch is able to detect intel edison os when using yocto linux (default os)

### CODE REVIEW

**Comment by alessandro-aglietti** [comment]
I don't know because travis fail but
python -m py_compile os_detect.py
doesn't do errors

maybe a problem with the build?

File: src/rospkg/os_detect.py [hunk 1]
```diff
@@ -430,6 +430,27 @@ def get_codename(self):
...
 
+
+class IntelEdisonYoctoDetect(OsDetector):
+    """
+    Detect Intel Edison Yocto Linux.
+    Edison uname -a
+    Linux myedison 3.10.17-poky-edison+ #1 SMP PREEMPT Fri Jun 19 12:06:40 CEST 2015 i686 GNU/Linux
+    """
+    def is_os(self):
+        return platform.release() == "3.10.17-poky-edison+"
```
Comment (tfoote) [inline src/rospkg/os_detect.py]: This detector should not be kernel version specific.

File: src/rospkg/os_detect.py [hunk 2]
```diff
@@ -430,6 +430,27 @@ def get_codename(self):
...
+
+    def get_version(self):
+        if self.is_os():
+            return platform.version()
+        raise OsNotDetected('called in incorrect OS')
+
+    def get_codename(self):
+        if self.is_os():
+            return platform.release()
```
Comment (tfoote) [inline src/rospkg/os_detect.py]: The release string as called out above is not equivalent to a code name.

**Comment by tfoote** [comment]
I believe there's a problem with python 3.2 and coverage: See @wjwwood's comment here: https://github.com/ros-infrastructure/rosdep/pull/412#issuecomment-143057022

Can you explain a little bit more about your use case and how you intend to use this. Putting things like the kernel version into the OS detection seems to be the wrong model. We don't want to require rereleasing rospkg for every new OS release.

File: src/rospkg/os_detect.py (hunk 2 above)
Comment (alessandro-aglietti) [inline src/rospkg/os_detect.py]: Sorry for that, I've update the pull request

**Comment by alessandro-aglietti** [comment]
Sorry for that, I've update the pull request.

The goal is support ros installation via pip on the Intel Edison Yocto Linux: https://en.wikipedia.org/wiki/Intel_Edison

**Comment by tfoote** [comment]
To support this use case the best would probably be to detect a class of the linux distro(aka yocto) and not the specific hardware. Since you can also run more standard debian/ubilinux on it. It would probably be reuseable for other yocto builds.

**Comment by dirk-thomas** [comment]
I will close this ticket for now since it has been stated that the detection should be based on the distro and not based on the hardware.

**Pull request closed without merging**